src/OptimalTransport.jl

```diff
@@ -467,32 +491,51 @@ where ``\\mathrm{entropicOT}^{\\epsilon}_{C}`` denotes the entropic optimal tran
 function sinkhorn_barycenter(
     mu_all, C_all, eps, lambda_all; tol=1e-9, check_marginal_step=10, max_iter=1000
 )
-    sums = sum(mu_all; dims=2)
+    sums = sum(mu_all; dims=1)
     if !isapprox(extrema(sums)...)
         throw(ArgumentError("Error: marginals are unbalanced"))
     end
-    K_all = [exp.(-C_all[i] / eps) for i in 1:length(C_all)]
+    # if batch_kernel = true, then compute all kernel reductions as matmul
+    batch_kernel = (C_all isa Matrix)
+    if batch_kernel
+        K_all = exp.(-C_all / eps)
+    else
+        K_all = [exp.(-C_all[i] / eps) for i in 1:length(C_all)]
+    end
```

Comment:
Can we again maybe compute everything in a vectorized form, i.e., use 3-dimensional arrays instead of vectors of arrays such that one can write
```suggestion
    K_all = @. exp(-C_all / eps)
```
?

Reply:
Ok about this, we can certainly do so. Unlike in the case of `sinkhorn_gibbs`, even using the 3-way arrays we would still need to handle the kernel reductions separately. I'm actually beginning to think that it would be quite uncommon to solve for barycenters with multiple cost matrices (this could correspond to the input measures technically living on different spaces). So perhaps we should just require a single cost matrix (so everything lives on a common space, as per the typical usage). Then we compute in batch.